src/DependencyInjection/Configuration.php

```diff
@@ -154,11 +154,15 @@ public function getConfigTreeBuilder(): TreeBuilder
                                 ->values(['authorization_code', 'implicit'])
                                 ->info('Grant type to use: authorization_code or implicit.')
                                 ->cannotBeEmpty()
-                                ->defaultValue(static function () {
-                                    @trigger_error('The "grant_type" configuration key has been deprecated since 5.3.0 and will be removed in 6.0.', E_USER_DEPRECATED);
+                                ->defaultValue('authorization_code')
+                                ->validate()
+                                    ->ifString()
```

Comment:
```suggestion
                                    ->always()
```

Reply:
:-1: Nop, cause we only want this deprecation to be set if the config key is set. If it's not set, the deprecation should be triggered